workflow/Utils/utils.go

```diff
@@ -138,22 +128,64 @@ func CreateHTTPClient() *http.Client {
 
 func CreateUrl(BaseUrl string, Port string, APIUrl string) (string, error) {
 
-	if !strings.HasPrefix(BaseUrl, "http://") && !strings.HasPrefix(BaseUrl, "https://") {
-		return "", errors.New("BaseUrl must start with http:// or https://")
-	}
+	finalURL := fmt.Sprintf("%s:%s%s", BaseUrl, Port, APIUrl)
+	return finalURL, nil
+}
+
+func EmailGenerator(orderID uint, orderStatus string) ([]byte, error) {
+	var subject, body string
+
+	switch strings.ToLower(orderStatus) {
+	case OrderPlaced:
+		subject = OrderPlacedSubject
+		body = fmt.Sprintf("Great news! Your order with Order ID %v has been successfully placed. Please wait for the restaurant to confirm your order. Thank you for choosing us.", orderID)
+
+	case Cancelled:
+		subject = CancelledSubject
+		body = fmt.Sprintf("We regret to inform you that your order with Order ID %v has been cancelled. If you have any questions, please contact support.", orderID)
 
-	if _, err := url.ParseRequestURI(Port); err != nil {
-		return "", fmt.Errorf("invalid Port: %v", err)
+	case Accepted:
+		subject = AcceptedSubject
+		body = fmt.Sprintf("Great news! Your order with Order ID %v has been confirmed. Thank you for choosing us.", orderID)
+
+	case Completed:
+		subject = CompletedSubject
+		body = fmt.Sprintf("Congratulations! Your order with Order ID %v has been successfully completed. We would appreciate it if you could leave a review. Thank you!", orderID)
+
+	case Delay:
+		subject = DelaySubject
+		body = fmt.Sprintf("We regret to inform you that your order with Order ID %v has been delayed because the restaurant has not accepted it yet.", orderID)
+
+	default:
+		log.Println("Invalid order status provided.")
+		return []byte{}, fmt.Errorf("invalid order status: %s", orderStatus)
 	}
 
-	baseURL, err := url.Parse(BaseUrl)
-	if err != nil {
-		return "", fmt.Errorf("invalid BaseUrl: %v", err)
+	message := []byte(subject + "\n" + body)
+	return message, nil
+}
+
+func ActivityOptions() workflow.ActivityOptions {
```

Comment:
method name

Reply:
i have changed the function name to GenerateEmail.